File: 其他/案例/init_demo/svn_pro/xmmq_app/lib/models/api/address.dart
class AddressModel {
  bool success;
  String errorCode;
  String errorDesc;
  String message;
  Data data;

  AddressModel(
      {this.success, this.errorCode, this.errorDesc, this.message, this.data});

  AddressModel.fromJson(Map<String, dynamic> json) {
    success = json['Success'];
    errorCode = json['ErrorCode'];
    errorDesc = json['ErrorDesc'];
    message = json['Message'];
    data = json['Data'] != null ? new Data.fromJson(json['Data']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['Success'] = this.success;
    data['ErrorCode'] = this.errorCode;
    data['ErrorDesc'] = this.errorDesc;
    data['Message'] = this.message;
    if (this.data != null) {
      data['Data'] = this.data.toJson();
    }
    return data;
  }
}

class Data {
  int pageIndex;
  int pageSize;
  int totalCount;
  List<ListObjects> listObjects;

  Data({this.pageIndex, this.pageSize, this.totalCount, this.listObjects});

  Data.fromJson(Map<String, dynamic> json) {
    pageIndex = json['PageIndex'];
    pageSize = json['PageSize'];
    totalCount = json['TotalCount'];
    if (json['ListObjects'] != null) {
      listObjects = new List<ListObjects>();
      json['ListObjects'].forEach((v) {
        listObjects.add(new ListObjects.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['PageIndex'] = this.pageIndex;
    data['PageSize'] = this.pageSize;
    data['TotalCount'] = this.totalCount;
    if (this.listObjects != null) {
      data['ListObjects'] = this.listObjects.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class ListObjects {
  String guid;
  String storeCustomerGuid;
  int addressType;
  String fullName;
  String addressLine;
  String postCode;
  String town;
  String city;
  String province;
  String phoneNumber;
  int iDCardType;
  String iDCardNumber;
  String country;
  int displayOrder;

  ListObjects(
      {this.guid,
      this.storeCustomerGuid,
      this.addressType,
      this.fullName,
      this.addressLine,
      this.postCode,
      this.town,
      this.city,
      this.province,
      this.phoneNumber,
      this.iDCardType,
      this.iDCardNumber,
      this.country,
      this.displayOrder});

  ListObjects.fromJson(Map<String, dynamic> json) {
    guid = json['Guid'];
    storeCustomerGuid = json['StoreCustomerGuid'];
    addressType = json['AddressType'];
    fullName = json['FullName'];
    addressLine = json['AddressLine'];
    postCode = json['PostCode'];
    town = json['Town'];
    city = json['City'];
    province = json['Province'];
    phoneNumber = json['PhoneNumber'];
    iDCardType = json['IDCardType'];
    iDCardNumber = json['IDCardNumber'];
    country = json['Country'];
    displayOrder = json['DisplayOrder'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['Guid'] = this.guid;
    data['StoreCustomerGuid'] = this.storeCustomerGuid;
    data['AddressType'] = this.addressType;
    data['FullName'] = this.fullName;
    data['AddressLine'] = this.addressLine;
    data['PostCode'] = this.postCode;
    data['Town'] = this.town;
    data['City'] = this.city;
    data['Province'] = this.province;
    data['PhoneNumber'] = this.phoneNumber;
    data['IDCardType'] = this.iDCardType;
    data['IDCardNumber'] = this.iDCardNumber;
    data['Country'] = this.country;
    data['DisplayOrder'] = this.displayOrder;
    return data;
  }
}

File: 其他/案例/init_demo/svn_pro/xiaomaimai_wechat/v2/pages/address/new-address/address.js
import webapi from '../../../utils/webapi'
import areaList from '../../../utils/area'

import util from "../../../utils/util";
const app = getApp();

Page({
  data: {
    address: {
      Guid: '',
      FullName: '',
      Province: '',
      City: '',
      Town: '',
      PostCode: '',
      PhoneNumber: '',
      AddressLine: '',
      IDCardNumber: '',
      IDCardType: 0,
      AddressType: 4,
    },
    isEditCard: false,
    showAreaList: false,
    areaList: areaList,
    isDefaultAddress: false,
    areas: '',
    errors: {}
  },
  onLoad(options) {
    const eventChannel = this.getOpenerEventChannel()
    eventChannel.emit('getAddress', (address) => {
      if (address) {
        this.setData({
          address: address,
          idcard: address.IDCardNumber.replace(/^([\s\S]{4})([\s\S]{11})([\s\S]*)$/g, '$1***********$3'),
          areas: `${address.Province}/${address.City}/${address.Town}`,
          isDefaultAddress: address.AddressType === 2
        });
      }
    });
  },
  formSubmit(e) {
    var adds = e.detail.value;
    const eventChannel = this.getOpenerEventChannel()
    if (this.validate(adds)) {
      const { address, isEditCard } = this.data;
      const postData = {
        ...address,
      }
      if (isEditCard) {
        postData.IDCardNumber = adds.idcard
      }
      address.StoreCustomerGuid = app.globalData.apiLoginInfo.StoreCustomerGuid;
      (postData.Guid === '' ?
        webapi.addStoreCustomerAddress(postData) :
        webapi.updateStoreCustomerAddress(postData))
        .then((res) => {
          if (res.Success) {
            eventChannel.emit('onLoadAddress', postData);
            wx.navigateBack({
              delta: 1
            });
          }
        });
    }
  },
  validate(data) {
    console.log(data);
    const errorData = {};
    const { isEditCard } = this.data;
    const { name, phone, areas, address, idcard } = data;

    if (name.length === 0) {
      errorData.FullName = '请输入收件人姓名';
    }
    if (phone.length === 0) {
      errorData.PhoneNumber = '输入收件人手机号';
    } else if (!/^\d{11}$/ig.test(data.phone)) {
      errorData.PhoneNumber = '输入正确的手机号';
    }
    if (areas.length === 0) {
      errorData.area = '请选择所在地区';
    }
    if (address.length === 0) {
      errorData.AddressLine = '请输入收件人详细地址';
    }

    if (isEditCard && idcard.length > 0 && !util.isCardId(idcard)) {
      errorData.IDCardNumber = '请输入正确的身份证号';
    }
    if (Object.keys(errorData).length === 0) {
      return true;
    } else {
      this.setData({
        errors: errorData
      })
      return false;
    }
  },
  onBlur(e) {
    this.setData({
      address: {
        ...this.data.address,
        [e.target.dataset.name]: e.detail.value
      },
      errors: {
        ...this.data.errors,
        [e.target.dataset.name]: ''
      }
    })
  },
  showPopup(e) {
    this.setData({
      showAreaList: true,
    });
  },
  onCloseAreaList(e) {
    this.setData({
      showAreaList: false,
    });
  },
  onAreaCofrim(e) {
    if (!e.detail.index.includes(0)) {
      const values = e.detail.values;
      this.setData({
        areas: values.map(item => item.name).join('/'),
        address: {
          ...this.data.address,
          Province: values[0].name,
          City: values[1].name,
          Town: values[2].name,
          PostCode: values[2].code
        },
        errors: {
          ...this.data.errors,
          area: ''
        }
      });
      this.onCloseAreaList();
    }
  },
  onChangeDefaultAddress(e) {
    this.setData({
      address: {
        ...this.data.address,
        AddressType: e.detail ? 2 : 4
      },
      isDefaultAddress: e.detail
    })
  },
  onChangeAddressLine(e) {
    this.setData({
      address: {
        ...this.data.address,
        AddressLine: e.detail
      },
    })
  },
  onChangeIDCard({ detail }) {
    this.setData({
      isEditCard: true,
      idcard: detail
    })
  }
})
